---
title: "HBEC Acute vs Repeated Red Oak Exposure: Secreted Protein Analysis (Between Groups and Time Points) - Pre Exposure Samples"
author: "Elise Hickman"
output: html_document
---

# Set up workspace

```{r message = FALSE, warning = FALSE}
# Clear global environment
rm(list=ls())

# Load packages
library(this.path) # for file path
library(janitor) # for data cleaning
library(openxlsx) # for writing out files
library(tidyverse) # for data organization and manipulation
library(patchwork) # for graphing
library(pheatmap) # for graphing
library(ggpubr) # for normality testing QQ plot panel
library(rstatix) # for statistical testing
library(factoextra) # for PCA 
library(ComplexUpset) # for upset plots
library(biomaRt) # for Uniprot to gene name conversion
library(geneSynonym) # for getting gene synonyms, install using devtools
library(RColorBrewer) # for plotting

# Redefine functions that may be masked
select <- dplyr::select
rename <- dplyr::rename
margin <- ggplot2::margin
count <- dplyr::count

# Set theme
theme_set(theme_bw())

# Set working directory
setwd(this.dir())
```

# Import and clean data

We are focusing only on the pre-exposure data because there is more consistency in the amount of time that the cells had been incubating with the media and because this also reflects the same time that the samples for RNAseq, with which we want to compare the results, were collected. 

Import data:
```{r} 
# MaxPlex
max <- read.xlsx("3_ProcessedData/ProcessedData_MaxPlex_Imputed_Pre.xlsx") %>%
  filter(!str_detect(sample_id, "Day1_"))

# CoreImmune
ci <-  read.xlsx("3_ProcessedData/ProcessedData_CoreImmune_Imputed_Pre.xlsx") %>%
  filter(!str_detect(sample_id, "Day1_"))
```

Combine dataframes:
```{r}
# Get proteins in the MaxPlex that are not part of the Core Immune panel
prots_unique_maxplex <- setdiff(colnames(max), colnames(ci)) 

# Filter and combine datasets
pre <- left_join(ci, max %>% select(c(sample_id, all_of(prots_unique_maxplex))), by = "sample_id")
```


# Normality testing

## Raw data

Shapiro-Wilk test:
```{r}
# Post
shapiro_res <- pre %>%
  pivot_longer(!sample_id, names_to = "protein", values_to = "value") %>%
  group_by(protein) %>%
  shapiro_test(value) %>%
  mutate(normal = ifelse(p < 0.05, F, T))

count(shapiro_res, normal)
```

Histograms and Q-Q Plots:
```{r}
# Make figure panel of histograms
hist <- ggplot(pre %>% pivot_longer(!sample_id, names_to = "protein", values_to = "value"), aes(value)) +
  geom_histogram(fill = "gray40", color = "black", binwidth = function(x) {(max(x) - min(x))/25}) +
  facet_wrap(~ protein, scales = "free") +
  labs(y = "# of Observations", x = "Value") +
  theme(axis.text = element_blank(),
        axis.title = element_text(size = 20))

png("6_Figures/2_Pre/Hist_Panel_Pre.png", width = 12, height = 10, units = "in", res = 300)
hist
invisible(dev.off())

# Make figure panel of Q-Q plots
qq <- ggqqplot(pre %>% pivot_longer(!sample_id, names_to = "protein", values_to = "value"), 
         x = "value", facet.by = "protein", ggtheme = theme_bw(), scales = "free")

png("6_Figures/2_Pre/QQ_Panel_Pre.png", width = 30, height = 30, units = "in", res = 300)
qq
invisible(dev.off())
```

## Log2 Transformed Data

Shapiro-Wilk test:
```{r}
# Post
shapiro_res_log2 <- pre %>%
  pivot_longer(!sample_id, names_to = "protein", values_to = "value") %>%
  mutate(value = log2(value + 1)) %>%
  group_by(protein) %>%
  shapiro_test(value) %>%
  mutate(normal = ifelse(p < 0.05, F, T))

count(shapiro_res_log2, normal)
```

Histograms and Q-Q Plots:
```{r}
# Make figure panel of histograms
hist_log2 <- ggplot(pre %>% 
         pivot_longer(!sample_id, names_to = "protein", values_to = "value") %>%
         mutate(value = log2(value + 1)), aes(value)) +
  geom_histogram(fill = "gray40", color = "black", binwidth = function(x) {(max(x) - min(x))/25}) +
  facet_wrap(~ protein, scales = "free") +
  labs(y = "# of Observations", x = "Value") +
  theme(axis.text = element_blank(),
        axis.title = element_text(size = 20))

png("6_Figures/2_Pre/Hist_Panel_Post_Log2.png", width = 12, height = 10, units = "in", res = 300)
hist_log2
invisible(dev.off())

# Make figure panel of Q-Q plots
qq_log2 <- ggqqplot(pre %>% 
           pivot_longer(!sample_id, names_to = "protein", values_to = "value") %>%
           mutate(value = log2(value + 1)), 
         x = "value", facet.by = "protein", ggtheme = theme_bw(), scales = "free")

png("6_Figures/2_Pre/QQ_Panel_Post_Log2.png", width = 30, height = 30, units = "in", res = 300)
qq_log2
invisible(dev.off())
```

Overall, data look much more normal with log2 transformation, so we will proceed with the log2 transformed data:
```{r}
pre_log2 <- pre %>%
  mutate(across(where(is.numeric), \(x) log2(x+1)))
```


# Outlier Detection (PCA)

One standard way to detect outliers is the criterion of being "more than 6 standard deviations away from the mean." [Source](https://privefl.github.io/blog/detecting-outlier-samples-in-pca/). We can create a scoring function to extract how many and which samples are more than a certain number of standard deviations from the mean.

```{r}
# Create a scoring funciton to detect PCA sample outliers. The input is PCA results data frame and the number of standard deviations for the cutoff. The output is outlier names. 
outlier_detection = function(pca_df, sd){

    # getting scores
    scores = pca_df$x
    
    # identifying samples that are > x standard deviations away from the mean
    outlier_indices = apply(scores, 2, function(x) which( abs(x - mean(x)) > (sd * sd(x)) )) 
    
    # Select first three PCs and collapse results
    outlier_indices <- outlier_indices[1:3] %>%
        Reduce(union, .)
    
    # getting sample names
    outliers = rownames(scores)[outlier_indices]
    
    return(outliers)
}
```


Prepare data frames and apply scoring function:
```{r}
# Create annotation dataframe
pre_key <- pre %>%
  select(sample_id) %>%
  separate(sample_id, into = c("donor", "exposure", "timepoint", NA), remove = FALSE) %>%
  mutate(exposure = recode(exposure, "Ctrl" = "Vehicle", "RO" = "Red Oak")) %>%
  mutate(exposure = factor(exposure, levels = c("Vehicle", "Red Oak"))) %>%
  mutate(timepoint = factor(timepoint, levels = c("Day1", "Day3", "Day5", "Day8", "Day10", "Day12"))) %>%
  column_to_rownames("sample_id")

# Run PCA
pca_pre <- prcomp(pre_log2 %>% column_to_rownames("sample_id"), center = TRUE, scale = TRUE)

# Call function with different standard deviation cutoffs
outliers_6 <- outlier_detection(pca_pre, 6)
outliers_5 <- outlier_detection(pca_pre, 5)
outliers_4 <- outlier_detection(pca_pre, 4)
outliers_3 <- outlier_detection(pca_pre, 3)

# Summarize outlier test results
data.frame(sd_cutoff = c(6, 5, 4, 3), n_outliers = c(length(outliers_6), length(outliers_5), length(outliers_4), length(outliers_3)))

# Visualize PCA - no annotations
pca_plot_pre <- fviz_pca_ind(pca_pre, 
             label = "none",
             pointsize = 3) +
  labs(title = "PCA: Pre Samples") +
  theme(axis.title = element_text(size = rel(1.1)),
      panel.border = element_rect(fill = NA, color = "black", linewidth = 0.3),
      panel.grid.minor = element_blank(),
      panel.grid.major = element_blank(), 
      plot.title = element_text(hjust = 0.5, size = rel(1.5), face = "bold"),
      legend.position = "none")

pca_plot_pre

# Visualize PCA - Donor
pca_plot_pre_donor <- fviz_pca_ind(pca_pre, 
             label = "none",
             pointsize = 3, 
             habillage = pre_key$donor, 
             mean.point = FALSE,
             addEllipses = TRUE) +
  labs(title = "PCA: By Donor (Pre)") +
  scale_color_manual(name = "Donor", values = c("#7ad151", "#22a884", "#2a788e", "#414487")) +
  scale_fill_manual(name = "Donor", values = c("#7ad151", "#22a884", "#2a788e", "#414487")) +
  scale_shape_manual(name = "Donor", values = c(15, 16, 17, 18)) + 
  theme(axis.title = element_text(size = rel(1.1)),
      panel.border = element_rect(fill = NA, color = "black", linewidth = 0.3),
      panel.grid.minor = element_blank(),
      panel.grid.major = element_blank(), 
      plot.title = element_text(hjust = 0.5, size = rel(1.5), face = "bold"),
      plot.margin = unit(c(0, 0.5, 0, 0), "in"))

pca_plot_pre_donor

# Visualize PCA - Exposure
pca_plot_pre_expo <- fviz_pca_ind(pca_pre, 
             label = "none",
             pointsize = 3, 
             habillage = pre_key$exposure, 
             mean.point = FALSE, 
             addEllipses = TRUE) +
  labs(title = "PCA: By Exposure (Pre)") +
  scale_color_manual(name = "Exposure", values = c("steelblue2", "gold1")) +
  scale_fill_manual(name = "Exposure", values = c("steelblue2", "gold1")) +
  scale_shape_manual(name = "Exposure", values = c(15, 16)) + 
  theme(axis.title = element_text(size = rel(1.1)),
      panel.border = element_rect(fill = NA, color = "black", linewidth = 0.3),
      panel.grid.minor = element_blank(),
      panel.grid.major = element_blank(), 
      plot.title = element_text(hjust = 0.5, size = rel(1.5), face = "bold"),
      plot.margin = unit(c(0, 0.5, 0, 0), "in"))

pca_plot_pre_expo

# Visualize PCA - Day
pca_plot_pre_day <- fviz_pca_ind(pca_pre, 
             label = "none",
             pointsize = 3, 
             habillage = pre_key$timepoint, 
             mean.point = FALSE,
             addEllipses = TRUE) +
  labs(title = "PCA: By Day (Pre)") +
  scale_color_manual(name = "Day", values = c("#f3e55d", "#fc9f07", "#e8602d", "#bc3754", "#82206c", "#470b6a")) +
  scale_fill_manual(name = "Day", values = c("#f3e55d", "#fc9f07", "#e8602d", "#bc3754", "#82206c", "#470b6a")) +
  scale_shape_manual(name = "Day", values = c(15, 16, 17, 18, 3, 13)) + 
  theme(axis.title = element_text(size = rel(1.1)),
      panel.border = element_rect(fill = NA, color = "black", linewidth = 0.3),
      panel.grid.minor = element_blank(),
      panel.grid.major = element_blank(), 
      plot.title = element_text(hjust = 0.5, size = rel(1.5), face = "bold"))

pca_plot_pre_day

# Make patchwork figure
pca_patch_pre <- pca_plot_pre_expo + pca_plot_pre_donor + pca_plot_pre_day

png("6_Figures/2_Pre/PCA_Panel_ByVariable_Pre.png", width = 15, height = 4, units = "in", res = 300)
pca_patch_pre
invisible(dev.off())

# Save off dataframes for interindividual analysis
save(file = '3_ProcessedData/pre_key.Rdata', pre_key)
save(file = '3_ProcessedData/pre_log2.Rdata', pre_log2)
```


# Statistical Testing

## T-Test

Run a t-test within each day to see differences in treatment versus vehicle on each day.
```{r}
ttest_res_pre <- pre_log2 %>% 
  pivot_longer(!sample_id, names_to = "protein", values_to = "value") %>%
  separate(sample_id, into = c("donor", "treatment", "day", "collection")) %>%
  mutate(day = factor(day, levels = c("Day1", "Day3", "Day5", "Day8", "Day10", "Day12"))) %>%
  group_by(protein, day) %>%
  t_test(value ~ treatment, paired = TRUE) %>%
  group_split(day) %>%
  bind_rows()
```

We also need to figure out the directionality of the changes by day. For this, we can calculate the fold changes (red oak divided by vehicle) for each donor, then average them for each protein, then take the log2 of the averaged fold change.

```{r}
avg_foldchange <- pre %>%
  pivot_longer(!sample_id, names_to = "protein", values_to = "value") %>%
  separate(sample_id, into = c("donor", "treatment", "day", "collection")) %>%
  mutate(day = factor(day, levels = c("Day1", "Day3", "Day5", "Day8", "Day10", "Day12"))) %>%
  group_by(day, protein, donor) %>%
  pivot_wider(names_from = "treatment", values_from = "value") %>%
  mutate(fold_change = RO/Ctrl) %>%
  ungroup() %>%
  group_by(day, protein) %>%
  summarise(avg_fold_change = mean(fold_change, na.rm = TRUE), .groups = "drop") %>%
  mutate(log2FC = log2(avg_fold_change)) %>%
  unite(day_protein, day, protein, sep = " ")
```

Pull proteins and days that we want fold change for and bind log2FC results.
```{r}
ttest_res_pre_organized <- ttest_res_pre %>% 
  unite(day_protein, day, protein, sep = " ", remove = FALSE) %>%
  filter(p < 0.05) %>%
  left_join(avg_foldchange, by = "day_protein")

ttest_res_summary <- ttest_res_pre_organized %>% 
  mutate(direction = case_when(
    log2FC > 0 ~ "increased",
    log2FC < 0 ~ "decreased",
    TRUE ~ "zero"
  )) %>%
  group_by(day, direction) %>%
  summarise(count = n(), .groups = "drop") %>%
  pivot_wider(id_cols = "day", names_from = "direction", values_from = "count") %>%
  mutate(total = decreased + increased)
```

Organize results, including summary statistics, p-values, and log2 fold changes, and cleaned protein names for supplement.  
```{r}
summary_stats_pre <- pre %>% 
  separate(sample_id, into = c("donor", "treatment", "day", NA)) %>%
  pivot_longer(-c(donor:day), names_to = "protein", values_to = "value") %>%
  group_by(protein, treatment, day) %>%
  summarise(
    mean = mean(value, na.rm = TRUE),
    sd = sd(value, na.rm = TRUE),
    .groups = "drop"
  ) %>%
  mutate(summary = sprintf("%.2f Â± %.2f", mean, sd)) %>%
  select(-c(mean, sd)) %>%
  pivot_wider(id_cols = c(protein, day), names_from = "treatment", values_from = "summary") %>%
  unite(day_protein, day, protein, sep = " ")

summary_full_pre <- ttest_res_pre %>%
  select(c(day, protein, p)) %>%
  unite(day_protein, day, protein, sep = " ", remove = FALSE) %>%
  left_join(avg_foldchange %>% select(c(day_protein, log2FC)), by = "day_protein") %>%
  mutate(p = round(p, digits = 4),
         log2FC = round(log2FC, digits = 2)) %>%
  left_join(summary_stats_pre, by = "day_protein") %>%
  relocate(c(Ctrl, RO), .before = "p") %>%
  select(-day_protein) %>%
  rename("Day" = "day", "Protein" = "protein", "Vehicle_Conc_Mean_SD" = "Ctrl", 
         "RedOak_Conc_Mean_SD" = "RO", "P_Value" = "p", "Log2FoldChange" = "log2FC") %>%
  arrange(Day, Protein)

# Read in protein keys
prot_key_maxplex <- read.xlsx("3_ProcessedData/ProcessedData_MaxPlex_ProteinKey.xlsx")
prot_key_coreimmune <- read.xlsx("3_ProcessedData/ProcessedData_CoreImmune_ProteinKey.xlsx")

# Filter protein keys to only proteins included in the final summary stats
prots_all <- summary_full_pre %>% pull("Protein") %>% unique()

# Get protein information from core immune, then maxplex, then join together. 
# Had some issues with differing spaces and separating symbols, which is why this script is so long
prot_description_key_coreimmune <- data.frame(Protein_Column_Name = prots_all) %>%
  mutate(Protein_Column_Name_2 = str_replace_all(Protein_Column_Name, "[^A-Za-z0-9]", "")) %>%
  left_join(prot_key_coreimmune %>%
              mutate(Protein_Column_Name_2 = str_replace_all(Protein_Column_Name, "[^A-Za-z0-9]", "")) %>%
              select(c(Uniprot_ID, Protein_Name, Protein_Column_Name_2)), 
            by = "Protein_Column_Name_2") %>% 
  na.omit()

prot_description_key_maxplex <- data.frame(Protein_Column_Name = prots_all) %>%
  mutate(Protein_Column_Name_2 = str_replace_all(Protein_Column_Name, "[^A-Za-z0-9]", "")) %>%
  left_join(prot_key_coreimmune %>% 
              mutate(Protein_Column_Name_2 = str_replace_all(Protein_Column_Name, "[^A-Za-z0-9]", "")) %>%
              select(c(Uniprot_ID, Protein_Name, Protein_Column_Name_2)),
            by = "Protein_Column_Name_2") %>%
  filter(is.na(Uniprot_ID)) %>%
  select(-c(Uniprot_ID, Protein_Name)) %>%
  left_join(prot_key_maxplex %>%
              mutate(Protein_Column_Name_2 = str_replace_all(Protein_Column_Name, "[^A-Za-z0-9]", "")) %>%
              select(c(Uniprot_ID, Protein_Name, Protein_Column_Name_2)), 
            by = "Protein_Column_Name_2")

prot_description_key <- rbind(prot_description_key_coreimmune, prot_description_key_maxplex) %>%
  select(-Protein_Column_Name_2)

# Get protein descriptions

## Get uniprot
prot_uniprots <- prot_description_key$Uniprot_ID

## Query ensembl
ensembl <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")

prot_info <- getBM(
  attributes = c("uniprotswissprot", "description"),
  filters = "uniprotswissprot",
  values = prot_uniprots,
  mart = ensembl
) %>% 
  mutate(description_clean = str_remove(description, "\\[.*")) %>%
  select(c(uniprotswissprot, description_clean)) %>%
  rename("Uniprot_ID" = "uniprotswissprot", "Description" = "description_clean")


# Merge results together and complete final cleaning
summary_full_pre_final <- summary_full_pre %>%
  rename("Protein_Column_Name" = "Protein") %>%
  left_join(prot_description_key, by = "Protein_Column_Name") %>%
  left_join(prot_info, by = "Uniprot_ID") %>%
  relocate(c(Uniprot_ID, Protein_Name, Protein_Column_Name, Description), .after = "Day") %>%
  mutate(Description = ifelse(Protein_Column_Name == "C1R", "complement C1r subcomponent", Description)) %>%
  mutate(Significant = ifelse(P_Value < 0.05, "Yes", "No"))

write.xlsx(summary_full_pre_final, "7_Tables/SummaryStats_Pre.xlsx")
```

# Visualization

## Upset Plots

### Significantly Increased

```{r}
# Prepare input dataframe
upsetprep_incr <- ttest_res_pre %>%
  filter(p < 0.05) %>% 
  unite(day_protein, day, protein, sep = " ", remove = FALSE) %>%
  left_join(avg_foldchange, by = "day_protein") %>%
  filter(log2FC > 0) %>%
  mutate(day = gsub("Day", "Day ", day)) %>%
  mutate(include = 1) %>%
  pivot_wider(id_cols = protein, names_from = "day", values_from = "include")
  
# Define color scheme
group_colors <- c("Day 3" = "#fc9f07","Day 5" = "#e8602d", 
                  "Day 8" = "#bc3754", "Day 10" = "#82206c", "Day 12" = "#470b6a")
```

```{r warning = FALSE}
# Make plot
upset_plot_incr <- upset(
  
  # Input data
  upsetprep_incr, 
  
  # Groups to plot
  c("Day 12", "Day 10", "Day 8", "Day 5", "Day 3"),
  
  # Don't order groups by set size
  sort_sets = FALSE,
  
  # X axis title
  name = "Day or Combination of Days",
  
  # Remove background stripes
  stripes='white', 
  
  # Controlling group colors
  matrix=(
        intersection_matrix(geom=geom_point(shape='circle filled', size=3))
        + scale_color_manual(
            values = group_colors,
            guide ='none'
        )
    ),
  
  ,
    queries=list(
        upset_query(set = 'Day 3', fill='#fc9f07'),
        upset_query(set = 'Day 5', fill='#e8602d'),
        upset_query(set = 'Day 8', fill = "#bc3754"),
        upset_query(set = 'Day 10', fill = "#82206c"),
        upset_query(set = 'Day 12', fill = "#470b6a")
    ),
  
  # Controlling appearance
  base_annotations = list(
    
    # Controlling appearance for main bar plot
        'Intersection size' = intersection_size(
            mapping = aes(fill = 'bars_color'),
            color = "black",
            fill = "#D93227", 
            bar_number_threshold = 100,
            text = element_text(vjust = -0.5)) + 
          labs(y = "Number of Proteins") +
          ylim(c(0, 20)) +
          scale_y_continuous(expand = expansion(mult = c(0, 0.2))) +
          theme(panel.border = element_rect(linewidth = 0.5, fill = NA))
  ),
  
  themes = upset_modify_themes(
    list(
      'intersections_matrix' = theme(axis.text.y = element_text(size = 10))
    )
  ),
  
  set_sizes = upset_set_size(mapping = aes(color = 'outline_color'),
                                 position = 'right') +
    ylab("Number of Significantly \nChanged Proteins") +
    scale_color_manual(values = c('outline_color' = "black"), guide = 'none') +
    geom_text(aes(label=..count..), hjust=-0.5, stat='count') +
    expand_limits(y = 30)
)

upset_plot_incr

png("6_Figures/2_Pre/Upset_Pre_Increased.png", width = 7, height = 5, units = "in", res = 1200)
upset_plot_incr
invisible(dev.off())
```

### Significantly Decreased

```{r}
# Prepare input dataframe
upsetprep_decr <- ttest_res_pre %>%
  filter(p < 0.05) %>% 
  unite(day_protein, day, protein, sep = " ", remove = FALSE) %>%
  left_join(avg_foldchange, by = "day_protein") %>%
  filter(log2FC < 0) %>%
  mutate(day = gsub("Day", "Day ", day)) %>%
  mutate(include = 1) %>%
  pivot_wider(id_cols = protein, names_from = "day", values_from = "include")
  
# Define color scheme
group_colors <- c("Day 3" = "#fc9f07","Day 5" = "#e8602d", 
                  "Day 8" = "#bc3754", "Day 10" = "#82206c", "Day 12" = "#470b6a")
```

```{r warning = FALSE}
# Make plot
upset_plot_decr <- upset(
  
  # Input data
  upsetprep_decr, 
  
  # Groups to plot
  c("Day 12", "Day 10", "Day 8", "Day 5", "Day 3"),
  
  # Don't order groups by set size
  sort_sets = FALSE,
  
  # X axis title
  name = "Day or Combination of Days",
  
  # Remove background stripes
  stripes='white', 
  
  # Controlling group colors
  matrix=(
        intersection_matrix(geom=geom_point(shape='circle filled', size=3))
        + scale_color_manual(
            values = group_colors,
            guide ='none'
        )
    ),
  
  ,
    queries=list(
        upset_query(set = 'Day 3', fill='#fc9f07'),
        upset_query(set = 'Day 5', fill='#e8602d'),
        upset_query(set = 'Day 8', fill = "#bc3754"),
        upset_query(set = 'Day 10', fill = "#82206c"),
        upset_query(set = 'Day 12', fill = "#470b6a")
    ),
  
  # Controlling appearance
  base_annotations = list(
    
    # Controlling appearance for main bar plot
        'Intersection size' = intersection_size(
            mapping = aes(fill = 'bars_color'),
            color = "black",
            fill = "#3A6DB0", 
            bar_number_threshold = 100,
            text = element_text(vjust = -0.5)) + 
          labs(y = "Number of Proteins") +
          ylim(c(0, 20)) +
          scale_y_continuous(expand = expansion(mult = c(0, 0.2))) +
          theme(panel.border = element_rect(linewidth = 0.5, fill = NA))
  ),
  
  themes = upset_modify_themes(
    list(
      'intersections_matrix' = theme(axis.text.y = element_text(size = 10))
    )
  ),
  
  set_sizes = upset_set_size(mapping = aes(color = 'outline_color'),
                                 position = 'right') +
    ylab("Number of Significantly \nChanged Proteins") +
    scale_color_manual(values = c('outline_color' = "black"), guide = 'none') +
    geom_text(aes(label=..count..), hjust=-0.5, stat='count') +
    expand_limits(y = 20)
)

upset_plot_decr

png("6_Figures/2_Pre/Upset_Pre_Decreased.png", width = 7, height = 5, units = "in", res = 1200)
upset_plot_decr
invisible(dev.off())
```


## Heatmap

Heatmap of fold change (treamtent/control) for any proteins changed on any of the days with statistical results overlaid. 

```{r}
# Get significant proteins on any day
ttest_sig_prots <- ttest_res_pre %>%
  filter(p < 0.05) %>%
  pull("protein")

# Prepare data for graphing by making a dataframe of only significant comparisons' log2FC 
# NAs were replaced with 0 so that they could be graphed properly
fc_hm <- avg_foldchange %>%
  select(-avg_fold_change) %>%
  separate(day_protein, into = c("day", "protein"), sep = " ") %>%
  filter(protein %in% all_of(ttest_sig_prots)) %>%
  mutate(protein = str_remove_all(protein, "[_.]")) %>%
  pivot_wider(id_cols = "day", names_from = "protein", values_from = "log2FC") %>%
  column_to_rownames("day") %>%
  t() %>% data.frame()

# Annotation hm
fc_hm_anno <- avg_foldchange %>%
  select(-avg_fold_change) %>%
  separate(day_protein, into = c("day", "protein"), sep = " ", remove = FALSE) %>%
  filter(protein %in% all_of(ttest_sig_prots)) %>%
  mutate(protein = str_remove_all(protein, "[_.]")) %>%
  left_join(ttest_res_pre %>% unite(day_protein, day, protein, sep = " ") %>% select(c(day_protein, p)), by = "day_protein") %>%
  mutate(anno = ifelse(p < 0.0001, "\u25CF\u25CF\u25CF\u25CF", ifelse(p < 0.001, "\u25CF\u25CF\u25CF", ifelse(p < 0.01, "\u25CF\u25CF", ifelse(p < 0.05, "\u25CF", ""))))) %>%
  select(c(day, protein, anno)) %>%
  pivot_wider(id_cols = "day", names_from = "protein", values_from = "anno") %>%
  column_to_rownames("day") %>%
  t() %>% data.frame()

# Define color palette
my_palette <- colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100)

# Define range of color axis
rg <- max(abs(fc_hm))

hm <- pheatmap(fc_hm,
         cluster_cols = FALSE,
         angle_col = c("0"),
         color = my_palette,
         breaks = seq(-rg, rg, length.out = 100),
         display_numbers = fc_hm_anno,
         fontsize_number = 10,
         fontsize_row = 8)

png("6_Figures/2_Pre/Heatmap_Sig_Pre.png", width = 5.5, height = 9, units = "in", res = 1200)
hm
invisible(dev.off())
```


## Integration with RNAseq data

Create conversion dataframes between protein names, uniprot ID, column names, and altered column names during the imputation process
```{r}
## MAXPLEX
# Read in unmodified data frame
prot_data_raw_max <- read.xlsx("1_RawData/RawData_SecretedProteins_MaxPlex.xlsx")

# Create key for protein info
prot_key_max <- prot_data_raw_max %>% 
  
  # Get only part of file that will assist with assigning protein names and LODs
  dplyr::slice(1:4) %>%
  select(17:279) %>%
  
  # Clean column names and formatting
  column_to_rownames("nELISA.ID:") %>%
  t() %>% data.frame() %>%
  dplyr::rename("Uniprot_ID" = "Uniprot.ID.", "Protein_Name" = "Protein.Name.", 
         "Upper_LOD" = "Upper.Limit.of.Detection..pg.ml.", "Lower_LOD" = "Lower.Limit.of.Detection..pg.ml.") %>%
  rownames_to_column("nELISA_ID") %>%
  
  # Create new column for more R compatible column names
  ## Remove parenthethicals from protein names
  mutate(Prot_Col_Name = sub("\\(.*", "", Protein_Name)) %>%
  ## Replace spaces with underscores
  mutate(Prot_Col_Name = sub(" ", "_", Prot_Col_Name)) %>%
  ## Remove underscores if they are the last character in the string
  mutate(Prot_Col_Name = sub("_$", "", Prot_Col_Name)) %>%
  ## Replace forward slash with period
  mutate(Prot_Col_Name = sub("/", ".", Prot_Col_Name)) 

# Manually add suffix to TGFB1 because there are duplicates in the Prot_Col_Name column
prot_key_max[222,6] <- "TGF-beta_1_LAP"
prot_key_max[223,6] <- "TGF-beta_1_total"

## CORE IMMUNE
# Read in unmodified data frame
prot_data_raw_ci <- read.xlsx("1_RawData/RawData_SecretedProteins_CoreImmune.xlsx")

# Create key for protein info
prot_key_ci <- prot_data_raw_ci %>% 
  
  # Get only part of file that will assist with assigning protein names and LODs
  dplyr::slice(1:4) %>%
  select(17:300) %>%
  
  # Clean column names and formatting
  column_to_rownames("nELISA.ID:") %>%
  t() %>% data.frame() %>%
  rename("Uniprot_ID" = "Uniprot.ID.", "Protein_Name" = "Protein.Name.", 
         "Upper_LOD" = "Upper.Limit.of.Detection..pg.ml.", "Lower_LOD" = "Lower.Limit.of.Detection..pg.ml.") %>%
  rownames_to_column("nELISA_ID") %>%
  
  # Create new column for more R compatible column names
  ## Remove parenthethicals from protein names
  mutate(Prot_Col_Name = sub("\\(.*", "", Protein_Name)) %>%
  ## Replace spaces with underscores
  mutate(Prot_Col_Name = sub(" ", "_", Prot_Col_Name)) %>%
  ## Remove underscores if they are the last character in the string
  mutate(Prot_Col_Name = sub("_$", "", Prot_Col_Name)) %>%
  ## Replace forward slash with period
  mutate(Prot_Col_Name = sub("/", ".", Prot_Col_Name)) 

# Manually add suffix to TGFB1 and CXCL12 because there are duplicates in the Prot_Col_Name column a
# Manually change name of first protein since having a number as the starting character isn't very R compatible
prot_key_ci[248,6] <- "TGF-beta_1_LAP"
prot_key_ci[249,6] <- "TGF-beta_1_total"
prot_key_ci[1,6] <- "TNFRSF9"
prot_key_ci[91,6] <- "CXCL12_alpha"
prot_key_ci[92,6] <- "CXCL12_beta"

# BOTH - replace dashes with periods in column names
prot_key_max <- prot_key_max %>%
  mutate(colname_clean = gsub("-", ".", Prot_Col_Name)) %>%
  mutate(colname_clean = gsub(" ", ".", colname_clean))

prot_key_ci <- prot_key_ci %>%
  mutate(colname_clean = gsub("-", ".", Prot_Col_Name)) %>%
  mutate(colname_clean = gsub(" ", ".", colname_clean))
```

Pull columns from dataframe of proteins that we are analyzing and match them to their uniprot IDs, first from the core immune panel, then from maxplex.
```{r}
# Separate datasets
prots_coreimmune_uniprot <- data.frame("colname_clean" = colnames(ci %>% select(-sample_id))) %>%
  left_join(prot_key_ci %>% select(c(colname_clean, Uniprot_ID)), by = "colname_clean")

prots_maxplex_uniprot <- data.frame("colname_clean" = prots_unique_maxplex) %>%
  left_join(prot_key_max %>% select(c(colname_clean, Uniprot_ID)), by = "colname_clean")

# Bind together
prots_uniprot <- bind_rows(prots_coreimmune_uniprot, prots_maxplex_uniprot)
```

Convert to gene symbols and find gene synonyms:
```{r}
# Vector of uniprot
prots_uniprot_vec <- prots_uniprot$Uniprot_ID

# Convert Uniprot to gene names
mart <- useEnsembl("ensembl","hsapiens_gene_ensembl")

prot_gene_key <- getBM(filter = "uniprot_gn_id", attributes = c("external_gene_name", "description", "uniprot_gn_id", "entrezgene_id"),
                values = prots_uniprot_vec, mart = mart)


# Pull synonyms for proteins of interest
gene_synonyms <- humanSyno(prot_gene_key$external_gene_name)

# Define initial list cleaning function
enframe_proteins <- function(x) { 
  
  df <- enframe(x, name = "NCBI_ID", value = "gene_synonym") %>%
    unnest(gene_synonym)
  
  return(df)
  
  }

# Collapse vector list into dataframe
gene_synonyms <- lapply(gene_synonyms, enframe_proteins)
gene_synonyms <- bind_rows(gene_synonyms, .id = "gene") %>%
  left_join(prot_gene_key, join_by("gene" == "external_gene_name"))
```

### Correlation between all proteins - Raw Counts

Prepare dataframe for running correlation:
```{r warning = FALSE}
# Read in data
counts_raw <- read.xlsx("2_SampleInfo/ProcessedData_BulkRNASeq_RawCounts_Cleaned.xlsx")

# Filter count data to only genes of interest
counts_raw_filtered <- counts_raw %>% 
  separate(gene, into = c("gene_cleaned", NA), sep = "_", remove = FALSE) %>%
  filter(gene_cleaned %in% all_of(gene_synonyms$gene_synonym)) %>%
  left_join(gene_synonyms %>% select(c(gene_synonym, uniprot_gn_id)), join_by("gene_cleaned" == "gene_synonym")) %>%
  distinct() %>%
  left_join(prots_uniprot, by = join_by("uniprot_gn_id" == "Uniprot_ID"))

# Filter count data to separate pre and post data, and to pivot data longer
counts_post_d1 <- counts_raw_filtered %>% 
  pivot_longer(-c(gene, gene_cleaned, uniprot_gn_id, colname_clean), names_to = "sample", values_to = "raw_counts_RNAseq") %>%
  filter(!str_detect(sample, "Ch")) %>%
  mutate(sample_matching = gsub("Ac", "Day1", sample)) %>%
  unite(sample_colname, sample_matching, colname_clean, sep = " ", remove = FALSE) %>%
  mutate(sample_colname = gsub("Day1", "Day3", sample_colname))

# Prepare nomic data to be joined to count data
prot_d3 <- pre_log2 %>% 
  separate(sample_id, into = c("donor", "treatment", "day", NA), sep = "_") %>%
  filter(day == "Day3") %>%
  mutate(treatment = recode(treatment, "Ctrl" = "CT")) %>%
  unite(sample_id, donor, treatment, day, sep = "_") %>%
  pivot_longer(!sample_id, names_to = "colname_clean", values_to = "log2_prot_expr") %>%
  unite(sample_colname, sample_id, colname_clean, sep = " ")

# Combine data
corrprep_d3 <- counts_post_d1 %>%
  left_join(prot_d3, by = "sample_colname")

corrprep_key_d3 <- corrprep_d3 %>%
  select(c(gene, gene_cleaned, colname_clean))
```

How many proteins were matched?
```{r}
nrow(counts_raw_filtered %>% filter(!duplicated(gene_cleaned)))
```

Run correlations
```{r}
corr_d3 <- corrprep_d3 %>%
  group_by(gene) %>%
  cor_test(raw_counts_RNAseq, log2_prot_expr, method = "spearman")

nrow(corr_d3 %>% filter(p < 0.05))
nrow(corr_d3 %>% filter(abs(cor) >= 0.6))
nrow(corr_d3 %>% filter(abs(cor) >= 0.6 & p < 0.05))
```

### Overlap in directionality and significance

What is the overlap between RNAseq results, nomic results, and correlation results?
```{r warning = FALSE}
# Read in data
deseq_ac <- read.xlsx("2_SampleInfo/DESeq2_Results_Acute.xlsx")

# Clean up dataframes 
deseq_ac_overlap <- deseq_ac %>%
  separate(gene, into = c("gene_cleaned", NA), sep = "_", remove = FALSE) %>%
  filter(gene_cleaned %in% gene_synonyms$gene_synonym) %>%
  select(c(gene, log2FoldChange, pvalue, padj)) %>%
  rename("log2FC_deseq" = "log2FoldChange", "p_deseq" = "pvalue", "padj_deseq" = "padj")

# First pull differentially expressed proteins
ttest_sig_d3 <- ttest_res_pre_organized %>%
  filter(day == "Day3") %>%
  rename("colname_clean" = "protein", "log2FC_prot" = "log2FC") %>%
  select(c(colname_clean, log2FC_prot))

# Then pull RNAseq results
deseq_ac_overlap_d3 <- deseq_ac_overlap %>%
  left_join(corrprep_key_d3, by = "gene") %>%
  distinct() %>%
  filter(colname_clean %in% all_of(ttest_sig_d3$colname_clean)) %>%
  left_join(ttest_sig_d3, by = "colname_clean")

count(deseq_ac_overlap_d3 %>% filter(p_deseq < 0.05))
count(deseq_ac_overlap_d3 %>% filter(padj_deseq < 0.05))
```

### Graph significant correlations

IL-24
```{r}
# Prepare data
corr_d3_il24 <- corrprep_d3 %>%
  filter(colname_clean == "IL.24") %>%
  separate(sample, into = c("Donor", "Exposure", NA), sep = "_") %>%
  mutate(Exposure = recode(Exposure, "CT" = "Vehicle", "RO" = "Red Oak")) %>% 
  mutate(Exposure = factor(Exposure, levels = c("Vehicle", "Red Oak")))

# Plot
IL24_corrplot <- ggplot(corr_d3_il24, aes(x = raw_counts_RNAseq, y = log2_prot_expr)) +
  geom_point(aes(color = Exposure, fill = Exposure, shape = Exposure), size = 4) + 
  scale_color_manual(name = "Exposure", values = c("steelblue2", "gold1")) +
  scale_fill_manual(name = "Exposure", values = c("steelblue2", "gold1")) +
  scale_shape_manual(name = "Exposure", values = c(15, 16)) +
  labs(x = "RNAseq Counts", y = bquote(~Log[2]~'Protein Concentration'), title = "Day 3 IL-24") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.title = element_text(face = "bold", hjust = 0.5),
        legend.position = "right",
        legend.justification = "top")

png("6_Figures/2_Pre/Corrplot_IL24_Day3.png", width = 4, height = 3, units = "in", res = 1200)
IL24_corrplot 
invisible(dev.off())

IL24_corrplot 
```

uPA_R:
```{r}
# Prepare data
corr_d3_uPAR <- corrprep_d3 %>%
  filter(gene == "PLAUR_1") %>%
  separate(sample, into = c("Donor", "Exposure", NA), sep = "_") %>%
  mutate(Exposure = recode(Exposure, "CT" = "Vehicle", "RO" = "Red Oak")) %>% 
  mutate(Exposure = factor(Exposure, levels = c("Vehicle", "Red Oak")))

# Plot
uPAR_corrplot <- ggplot(corr_d3_uPAR, aes(x = raw_counts_RNAseq, y = log2_prot_expr)) +
  geom_point(aes(color = Exposure, fill = Exposure, shape = Exposure), size = 4) + 
  scale_color_manual(name = "Exposure", values = c("steelblue2", "gold1")) +
  scale_fill_manual(name = "Exposure", values = c("steelblue2", "gold1")) +
  scale_shape_manual(name = "Exposure", values = c(15, 16)) +
  labs(x = "RNAseq Counts", y = bquote(~Log[2]~'Protein Concentration'), title = "Day 3 uPA-R") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.title = element_text(face = "bold", hjust = 0.5),
        legend.position = "right",
        legend.justification = "top")

png("6_Figures/2_Pre/Corrplot_uPAR_Day3.png", width = 4, height = 3, units = "in", res = 1200)
uPAR_corrplot
invisible(dev.off())

uPAR_corrplot
```

